### ISSUE TITLE
Compatibility with pytorch 1.5.0

### ISSUE THREAD

**Issue by bkmi** [e3nn/e3nn#74]
Here is some of the summary when using `pytest` with `pytorch==1.5.0`

```
FAILED equivariance_test.py::Tests::test_parity_gated_block_parity - RuntimeError: view size is not compatible with input tensor's size and stride (at...
FAILED equivariance_test.py::Tests::test_parity_gated_block_parity_mod - RuntimeError: view size is not compatible with input tensor's size and stride...
FAILED equivariance_test.py::Tests::test_parity_rotation_gated_block_parity - RuntimeError: view size is not compatible with input tensor's size and s...
FAILED equivariance_test.py::Tests::test_parity_rotation_gated_block_parity_mod - RuntimeError: view size is not compatible with input tensor's size a...
FAILED o3_test.py::Tests::test_sh_cuda_single - TypeError: 'int' object is not iterable
FAILED tensor_product_test.py::Tests::test_elementwise_tensor_product - RuntimeError: view size is not compatible with input tensor's size and stride ...
FAILED tensor_product_test.py::Tests::test_tensor_product - RuntimeError: view size is not compatible with input tensor's size and stride (at least on...
FAILED tensor_product_test.py::Tests::test_tensor_product_norm - RuntimeError: view size is not compatible with input tensor's size and stride (at lea...
```

Therefore tests are incompatible. I expect other things will be too since it seems like pytorch got more strict about views somehow.

**Comment by mariogeiger** [comment]
All right I will change the versions of pytorch in the CI and fix all the
issues

**Comment by mariogeiger** [comment]
Maybe we can write somewhere what version on pytorch we used while
developing the code

**Comment by mariogeiger** [comment]
solved in cf6ae0e59b40efed9bb9ff36933d4f6fa743ce4f

The problem what due to a change of `einsum` apparently 1.5.0 outputs more often non contiguous tensors. Which is a good sign of optimization I suppose.
I replaced all the `.view` in `.reshape`. I recommend to always use `.reshape`
